core/services/relay/evm/functions/coordinator_v2.go

```diff
@@ -0,0 +1,146 @@
+package functions
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/functions/generated/functions_coordinator"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	evmRelayTypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+type CoordinatorV2 struct {
+	address common.Address
+
+	client    client.Client
+	logPoller logpoller.LogPoller
+	lggr      logger.Logger
+}
+
+func NewCoordinatorV2(address common.Address, client client.Client, logPoller logpoller.LogPoller, lggr logger.Logger) *CoordinatorV2 {
+	return &CoordinatorV2{
+		address:   address,
+		client:    client,
+		logPoller: logPoller,
+		lggr:      lggr,
+	}
+}
+
+func (c *CoordinatorV2) Address() common.Address {
+	return c.address
+}
+
+func (c *CoordinatorV2) RegisterFilters() error {
+	if (c.address == common.Address{}) {
+		return nil
+	}
+
+	return c.logPoller.RegisterFilter(
+		logpoller.Filter{
+			Name: logpoller.FilterName("FunctionsLogPollerWrapper", c.address.String(), "-v", "2"),
```

Comment:
Move string to a const used by both files. Also you could encode "-v2" in the prefix.